diff --git a/src/Main.hs b/src/Main.hs
index 959ab8e..d3b6077 100644
--- a/src/Main.hs
+++ b/src/Main.hs
@@ -95,7 +95,12 @@ pathToText path =
     underneath `/nix/store`, but this is the overwhelmingly common use case
 -}
 derivationName :: FilePath -> Text
-derivationName = Data.Text.dropEnd 4 . Data.Text.drop 44 . pathToText
+derivationName p =
+    if Data.Text.isPrefixOf "nixos-system" s
+      then "nixos-system"
+      else s
+  where
+    s = Data.Text.dropEnd 4 . Data.Text.drop 44 . pathToText $ p
 
 -- | Group input derivations by their name
 groupByName :: Map FilePath (Set Text) -> Map Text (Map FilePath (Set Text))
