diff --git a/buildbot/reporters/words.py b/master/buildbot/reporters/words.py
index a65147b..bf44118 100644
--- a/buildbot/reporters/words.py
+++ b/buildbot/reporters/words.py
@@ -550,14 +550,15 @@ class Contact(service.AsyncService):
 
         if self.useRevisions:
             revisions = yield self.getRevisionsForBuild(build)
-            r = "Hey! build %s containing revision(s) [%s] is complete: %s" % \
+            r = "Build %s containing revision(s) [%s] is complete: %s" % \
                 (builderName, ','.join(revisions), results[0])
         else:
-            r = "Hey! build %s #%d is complete: %s" % \
+            r = "Build %s #%d is complete: %s" % \
                 (builderName, buildNumber, results[0])
 
         r += ' [%s]' % maybeColorize(build['state_string'],
                                      results[1], self.useColors)
+        r += " - %s" % self.master.status.getURLForBuild(builder['builderid'],buildNumber)
         self.send(r)
 
         # FIXME: where do we get the list of changes for a build ?
@@ -622,14 +623,15 @@ class Contact(service.AsyncService):
         results = self.getResultsDescriptionAndColor(build['results'])
         if self.useRevisions:
             revisions = yield self.getRevisionsForBuild(build)
-            r = "Hey! build %s containing revision(s) [%s] is complete: %s" % \
+            r = "Build %s containing revision(s) [%s] is complete: %s" % \
                 (builder_name, ','.join(revisions), results[0])
         else:
-            r = "Hey! build %s #%d is complete: %s" % \
+            r = "Build %s #%d is complete: %s" % \
                 (builder_name, buildnum, results[0])
 
         r += ' [%s]' % maybeColorize(build['state_string'],
                                      results[1], self.useColors)
+        r += " - %s" % self.master.status.getURLForBuild(builder['builderid'],buildNumber)
         self.send(r)
 
         # FIXME: where do we get the base_url? Then do we use the build Link to
