--- ./configure.in	2016-06-27 18:36:06.640779048 +0200
+++ ./configure.in	2016-06-27 18:34:53.968803854 +0200
@@ -119,11 +119,11 @@
      ;;
   *)
      AC_MSG_RESULT($withval)
-     if test -f $withval/pcap.h -a -f $withval/libpcap.a; then
+     if test -f $withval/include/pcap.h -a -f $withval/lib/libpcap.so; then
         owd=`pwd`
         if cd $withval; then withval=`pwd`; cd $owd; fi
-	PCAPINC="-I$withval -I$withval/bpf"
-	PCAPLIB="-L$withval -lpcap"
+	PCAPINC="-I$withval/include -I$withval/include/bpf"
+	PCAPLIB="-L$withval/lib -lpcap"
      else
         AC_ERROR(pcap.h or libpcap.a not found in $withval)
      fi
@@ -230,7 +230,7 @@
         if cd $withval; then withval=`pwd`; cd $owd; fi
         EVENTINC="-I$withval"
         EVENTLIB="-L$withval -levent"
-     elif test -f $withval/include/event.h -a -f $withval/lib/libevent.a; then
+     elif test -f $withval/include/event.h -a -f $withval/lib/libevent.so; then
         owd=`pwd`
         if cd $withval; then withval=`pwd`; cd $owd; fi
 	EVENTINC="-I$withval/include"
@@ -354,12 +354,12 @@
      ;;
   *)
      AC_MSG_RESULT($withval)
-     if test -f $withval/readline/readline.h -a -f $withval/libreadline.a; then
+     if test -f $withval/include/readline/readline.h -o -f $withval/lib/libreadline.so; then
         owd=`pwd`
         if cd $withval; then withval=`pwd`; cd $owd; fi
 	AC_DEFINE(HAVE_LIBREADLINE, 1, [Define if you have libreadline])
-        EDITINC="-I$withval"
-        EDITLIB="-L$withval -lreadline"
+        EDITINC="-I$withval/include"
+        EDITLIB="-L$withval/lib -lreadline"
      else
         AC_ERROR(readline/readline.h or libreadline.a not found in $withval)
      fi
