diff --git a/nichtparasoup.py b/nichtparasoup.py
index 9da9a2b..833ca71 100755
--- a/nichtparasoup.py
+++ b/nichtparasoup.py
@@ -211,7 +211,7 @@ def cache_fill_loop():
                     try:
                         sources[crawler][site].crawl()
                         info = Crawler.info()
-                    except Exception, e:
+                    except Exception as e:
                         logger.error("Error in crawler %s - %s: %s" % (crawler, site, e))
                         break
 
